lib/api/core/models/repositories/repository.js

```diff
@@ -0,0 +1,258 @@
+var
+  _ = require('lodash'),
+  q = require('q'),
+  RequestObject = require('../requestObject'),
+  ResponseObject = require('../responseObject');
+
+function Repository (kuzzle, options) {
+  this.collection = options.collection;
+  this.ttl = options.ttl || kuzzle.config.repositories.cacheTTL;
+  this.ObjectConstructor = options.ObjectConstructor;
+
+  this.readEngine = options.readEngine || kuzzle.services.list.readEngine;
+  this.writeEngine = options.writeEngine || kuzzle.services.list.writeEngine;
+  this.cacheEngine = options.cacheEngine || kuzzle.services.list.internalCache;
+}
+
+/**
+ *
+ * @param {string} id
+ * @returns {Promise} resolves on a new ObjectConstructor()
+ */
+Repository.prototype.loadOneFromDatabase = function (id) {
+  var
+    deferred = q.defer(),
+    requestObject,
+    result;
+
+  requestObject = new RequestObject({
+    controller: 'read',
+    action: 'get',
+    requestId: 'foo',
+    collection: this.collection,
+    body: {
+      _id: id
+    }
+  });
+
+  this.readEngine.get(requestObject)
+    .then(function (response) {
+      if (response.data) {
+        result = new this.ObjectConstructor();
+        deferred.resolve(this.hydrate(result, response));
+      }
+      else {
+        deferred.resolve(null);
+      }
+    }.bind(this))
+    .catch(function (error) {
+      if (error.status === 404) {
+        // no content found, we return null without failing
+        deferred.resolve(null);
+      }
+      else {
+        deferred.reject(error);
+      }
+    });
+
+  return deferred.promise;
+};
+
+/**
+ *
+ * @param ids
+ * @returns {*|promise}
+ */
+Repository.prototype.loadMultiFromDatabase = function (ids) {
+  var
+    deferred = q.defer(),
+    requestObject,
+    promises;
+
+  requestObject = new RequestObject({
+    controller: 'read',
+    action: 'mget',
+    requestId: 'foo',
+    collection: this.collection,
+    body: {
+      ids: ids
+    }
+  });
+
+  this.readEngine.mget(requestObject)
```

Comment:
Indeed, for the time being, the index is not taken into account.

On the role level, you can already define some permissions on an index but it actually won't be used and you need a wildcard entry for the time being.